lib/paranoia_helper.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+module ParanoiaHelper
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    private
+
+    # This method generates an instance method with the name: associated_sym.to_s.pluralize.
+    #
+    # Example:
+    #   class User < ActiveRecord::Base
+    #     belongs_to :role
+    #     collection_for_edit :role
+    #   end
+    #
+    # If the class defines a scope with a custom name, then that can be passed in with the options hash with the key 'name_prefix'
+    # like so:
+    #   class Store < ActiveRecord::Base
+    #     belongs_to :sales_person, class_name: User.to_s
+    #     collection_for_edit :sales_person, nil, name_prefix: :salesmen
+    #     collection_for_edit :sales_person, -> { User.salesmen }, name_prefix: :salesmen   # This is equivalent to the above
+    #   end
+    #
+    # The method will evaluate the scope_block if specified to get all the values for the edit dropdown. If the scope_block is nil,
+    # the method will try to deduce it from the association. Again, if there is a scope defined for the association class with the
+    # same name as option 'name_prefix' or the pluralized version of the associated symbol, then that will be used to fetch all
+    # records for the dropdown
+    #
+    def collection_for_edit(associated_sym, scope_block = nil, options = {})
```

Comment:
There's a lot going on inside of this method. Given, this codebase is legacy, but with this helper we have a chance to make things more readable. Are we able to use some private methods and reduce the ABC here?